internal/resources/k6/resource_project_allowed_load_zones.go

```diff
@@ -0,0 +1,233 @@
+	ctx = context.WithValue(ctx, k6.ContextAccessToken, r.config.Token)
+	_, httpResp, err := r.client.ProjectsAPI.ProjectsRetrieve(ctx, projectID).
+		XStackId(r.config.StackID).
+		Execute()
+
+	if httpResp != nil && httpResp.StatusCode == http.StatusNotFound {
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error reading k6 project",
+			fmt.Sprintf("Could not read k6 project with id %d: %v", projectID, err),
+		)
+		return
+	}
+
+	// Get allowed load zones
+	allowedZones, err := getProjectAllowedLoadZones(ctx, r.client, r.config, projectID)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error reading allowed load zones",
+			"Could not read allowed load zones for k6 project: "+err.Error(),
+		)
+		return
+	}
+
+	// Convert to types.List
+	var zoneValues []attr.Value
+	for _, zone := range allowedZones {
+		zoneValues = append(zoneValues, types.StringValue(zone))
+	}
+	state.AllowedLoadZones, diags = types.ListValue(types.StringType, zoneValues)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Set refreshed state
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update updates the resource and sets the updated Terraform state on success.
+func (r *projectAllowedLoadZonesResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// Retrieve values from plan
+	var plan projectAllowedLoadZonesResourceModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectID := plan.ProjectID.ValueInt32()
+
+	// Get load zones from plan
+	var loadZones []string
+	diags = plan.AllowedLoadZones.ElementsAs(ctx, &loadZones, false)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Update allowed load zones
+	err := setProjectAllowedLoadZones(ctx, r.client, r.config, projectID, loadZones)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error updating allowed load zones",
+			"Could not update allowed load zones for k6 project: "+err.Error(),
+		)
+		return
+	}
+
+	// Set state to fully populated data
+	diags = resp.State.Set(ctx, plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Delete deletes the resource and removes the Terraform state on success.
+func (r *projectAllowedLoadZonesResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	// Retrieve values from state
+	var state projectAllowedLoadZonesResourceModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectID := state.ProjectID.ValueInt32()
+
+	// Clear allowed load zones (set to empty list)
+	err := setProjectAllowedLoadZones(ctx, r.client, r.config, projectID, []string{})
```

Comment:
We should probably handle the 404 response here in case the related project is already removed.